avocado/plugins/messages.py

```diff
@@ -0,0 +1,123 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Red Hat Inc. 2021
+# Authors: Jan Richter <jarichte@redhat.com>
+
+"""
+Message resolver for runner messages
+"""
+
+import abc
+import os
+
+
+class MessageProcessor:
+    """Entry point for handling all types of messages"""
+
+    def __init__(self):
+        self._handlers = [StartMessageHandler(),
+                          RunningMessageProcessor(),
+                          FinishMessageHandler()]
+
+    def process_message(self, message, task, job):
+        for handler in self._handlers:
+            if handler.handle(message, task, job):
+                break
+
+
+class MessageHandler(metaclass=abc.ABCMeta):
+    """Base interface for resolving runner messages.
+
+    This is the interface a job uses to deal with messages form runners.
+    """
+
+    @abc.abstractmethod
+    def handle(self, message, task, job):
+        """Handle message from runner.
+
+        :param message: message from runner.
+        :type message: dict
+        :param task: runtime_task which message is related to
+        :type task: :class:`avocado.core.nrunner.Task`
+        :param job: job which task is related to
+        :type job: :class: `avocado.core.job.Job`
+        """
+
+
+class StartMessageHandler(MessageHandler):
+
+    def handle(self, message, task, job):
+        if message['status'] == 'started':
+            base_path = os.path.join(job.logdir, 'test-results')
```

Comment:
Ideally all information related to job like basepath, could be retrieved from the job class (inside the job class).

Also since this is returning the test-results path, I would rename this property too:

```python
@property
def test_results_path(self):
   return os.path.join(self.logdir, 'test-results')
```

Reply:
Are you proposing to add this property to the `Job` class?

Reply:
Yes.